ttg/ttg/terminal.h

```diff
@@ -11,8 +11,66 @@
 #include "ttg/util/trace.h"
 #include "ttg/util/demangle.h"
 #include "ttg/world.h"
+#include "boost/callable_traits.hpp"
 
 namespace ttg {
+  namespace detail {
+
+    /* Wraps any key,value data structure.
+     * Elements of the data structure can be accessed using get method, which calls the at method of the Container.
+     * keyT - taskID
+     * valueT - Value type of the Container
+    */
+    template<typename keyT, typename valueT>
+    struct ContainerWrapper {
+      std::function<valueT (keyT const& key)> get = nullptr;
+      std::function<size_t (keyT const& key)> owner = nullptr;
+
+      ContainerWrapper() = default;
+
+      template<typename T, typename mapperT, typename keymapT, std::enable_if_t<!std::is_same<std::decay_t<T>,
+                                                          ContainerWrapper>{}, bool> = true>
+        //Store a pointer to the user's container in std::any, no copies
+        ContainerWrapper(T &t, mapperT &&mapper,
+                         keymapT &&keymap) : get([&t,&mapper](keyT const &key) {
```

Comment:
We should not capture `mapper` by reference. If we do, this code snippet will end up with a dangling reference:
```
ContainerWrapper(vector, [](const Key& key){ return key.x; }, ...);
```

The lambda may be destroyed after the call, it's a temporary after all.

One solution would be:
```suggestion
                         keymapT &&keymap) : get([&t,mapper = std::forward<mapperT>(mapper)](keyT const &key) {
```

Here mapper is caught by value (which is what we want, and it's typically not expensive). 

The same is true for the `owner` below.